src/lib/database/daos/trick.ts

```diff
@@ -393,21 +394,144 @@ export class Trick implements DbObject {
+        if (trick.variationOf) {
+          const newVariation = trick.variationOf.filter(
+            (primaryKey) => !primaryKeysMatch(primaryKey, this.primaryKey)
+          );
+          if (trick.variationOf.length !== newVariation.length) {
+            this.db.tricks.update([trick.id, trick.trickStatus], { variationOf: newVariation });
+          }
+        }
+      });
+
+      // Remove trick from any comobs containing it
+      const allCombos = await this.db.combos.toArray();
+      allCombos.forEach((combo) => {
+        const newComboTricks = combo.tricks.filter(
+          (trick) => trick[0] != this.primaryKey[0] || trick[1] != this.primaryKey[1]
+        );
+        if (newComboTricks.length !== combo.tricks.length) {
+          this.db.combos.update([combo.id, combo.comboStatus], { tricks: newComboTricks });
+        }
+      });
+    });
+  }
+
+  public async updateStatusPersistent(
+    status: 'official' | 'archived' | 'userDefined'
+  ): Promise<void> {
+    if (this.#modified.deleted) {
+      return;
     }
+
+    return this.db.transaction('rw', this.db.tricks, this.db.metadata, this.db.combos, async () => {
+      const originalPrimaryKey = this.primaryKey;
+
+      // Check for id collision and generate new id if necessary
+      const allIdsOfTargetStatus: number[] = (await this.db.tricks.toArray())
+        .filter((trick) => trick.trickStatus === status)
+        .map((trick) => trick.id);
+      let newId = originalPrimaryKey[0];
+      if (allIdsOfTargetStatus.includes(originalPrimaryKey[0])) {
+        newId = Math.max(...allIdsOfTargetStatus) + 1;
+      }
+      const newPrimaryKey = [newId, status];
+
+      // Update trick itself
+      this.db.tricks.update(originalPrimaryKey, { id: newId, trickStatus: status });
+
+      // Update metadata
+      this.db.metadata.update([...originalPrimaryKey, 'Trick'], {
+        id: newId,
+        entityStatus: status,
+      });
+
+      // Update recommended prerequisites
+      const allTricks = await this.db.tricks.toArray();
+      allTricks.forEach((trick) => {
+        if (!trick.recommendedPrerequisites) {
+          return;
+        }
+        if (!trick.recommendedPrerequisites.some((e) => primaryKeysMatch(e, originalPrimaryKey))) {
+          return;
+        }
+
+        const newPrerequisites = trick.recommendedPrerequisites.map((prerequisitePrimaryKey) => {
+          return primaryKeysMatch(prerequisitePrimaryKey, originalPrimaryKey)
+            ? newPrimaryKey
+            : prerequisitePrimaryKey;
+        });
+        this.db.tricks.update([trick.id, trick.trickStatus], {
+          recommendedPrerequisites: newPrerequisites,
+        });
+      });
+
+      // Update other tricks variationOf
+      allTricks.forEach((trick) => {
+        if (!trick.variationOf) {
+          return;
+        }
+        if (!trick.variationOf.some((e) => primaryKeysMatch(e, originalPrimaryKey))) {
+          return;
+        }
```

Comment:
```suggestion
        if ((!trick.variationOf || !trick.variationOf.some((e) => primaryKeysMatch(e, originalPrimaryKey))) {
          return;
        }
```

Reply:
It makes for a fairly long line and might decrease readability by having one long if-statement which is harder to understand at a glance. But then also this functionally not very relevant (though necessary) section takes up less space this way. So I went with your suggestion and condensed the two ifs down to one. Same goes for your next comment.